src/middleware/authorisation.coffee

```diff
@@ -19,134 +15,57 @@ os = require 'os'
+    return authFunc channel, ctx
 
 exports.authorise = (ctx, done) ->
-  utils.getAllChannels (err, channels) ->
-    approvedChannels = []
-
-    for channel in channels
-      do (channel) ->
-        pat = new RegExp channel.urlPattern
-        # if url pattern matches
-        if pat.test ctx.request.path
-          matchedRoles = {}
-          allowedClient = false
-          if ctx.authenticated?
-            # used by messageStore
-            ctx.authenticated.ip = ctx.ip
-
-            if ctx.authenticated.roles?
-              matchedRoles = channel.allow.filter (element) ->
-                return (ctx.authenticated.roles.indexOf element) isnt -1
-            if ((channel.allow.indexOf ctx.authenticated.clientID) isnt -1)
-              allowedClient = true
-          else
-            # used by messageStore
-            ctx.authenticated =
-              ip: ctx.ip
-
-          # if the user has a role that is allowed or their username is allowed specifically
-          requireWhitelistCheck = channel.whitelist.length > 0
-          isWhiteListed = not requireWhitelistCheck or ((channel.whitelist.indexOf ctx.ip) isnt -1)
 
-          if isWhiteListed and ((matchedRoles.length > 0) or allowedClient or ((channel.authType == 'public') is true))
-            # authorisation success, now check if content type matches
-            if channel.matchContentTypes and channel.matchContentTypes.length > 0
-              if ctx.request.header and ctx.request.header['content-type']
-                ct = extractContentType ctx.request.header['content-type']
-                if (channel.matchContentTypes.indexOf ct) is -1
-                  # deny access to channel if the content type doesnt match
-                  return #(continue loop)
-              else
-                # deny access to channel if the content type isnt set
-                return #(continue loop)
-
-            # now check that the status is 'enabled' and if the message content matches
-            if isChannelEnabled(channel) and matchContent(channel, ctx.body)
-              approvedChannels.push channel
-
-    if approvedChannels.length > 0
-      # authorisation succeeded
-      logger.debug "Channels that match request: #{(approvedChannels.map (c) -> c.name)}"
-      ctx.authorisedChannel = channelWithHighestPriority approvedChannels
-      logger.info "The request, '" + ctx.request.path + "' is authorised to access " + ctx.authorisedChannel.name
-    else
-      # authorisation failed
-      ctx.response.status = 401
-      if config.authentication.enableBasicAuthentication
-        ctx.set "WWW-Authenticate", "Basic"
-      logger.info "The request, '" + ctx.request.path + "', is not authorised to access any channels."
-      auditing.sendAuditEvent genAuthAudit(ctx.ip), -> logger.debug 'Processed nodeAuthentication audit'
+  channel = ctx.matchingChannel
+  if channel? and (channel.authType == 'public' || isAuthorised(channel, ctx))
```

Comment:
I'm not sure if it should be. A public endpoint is accessible to anyone isn't it? The whitelist works with private channels that are authenticated by IP only.